standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -850,32 +977,34 @@ public Database getDatabase(String name) throws NoSuchObjectException {
       ex = e;
     }
     if (db == null) {
-      LOG.warn("Failed to get database {}, returning NoSuchObjectException", name, ex);
+      LOG.warn("Failed to get database {}.{}, returning NoSuchObjectException",
+          catalogName, name, ex);
       throw new NoSuchObjectException(name + (ex == null ? "" : (": " + ex.getMessage())));
     }
     return db;
   }
 
-  public Database getDatabaseInternal(String name) throws MetaException, NoSuchObjectException {
-    return new GetDbHelper(name, true, true) {
+  public Database getDatabaseInternal(String catalogName, String name)
+      throws MetaException, NoSuchObjectException {
+    return new GetDbHelper(catalogName, name, true, true) {
       @Override
       protected Database getSqlResult(GetHelper<Database> ctx) throws MetaException {
-        return directSql.getDatabase(dbName);
+        return directSql.getDatabase(catalogName, dbName);
       }
 
       @Override
       protected Database getJdoResult(GetHelper<Database> ctx) throws MetaException, NoSuchObjectException {
-        return getJDODatabase(dbName);
+        return getJDODatabase(catalogName, dbName);
       }
     }.run(false);
    }
 
-  public Database getJDODatabase(String name) throws NoSuchObjectException {
+  public Database getJDODatabase(String catName, String name) throws NoSuchObjectException {
     MDatabase mdb = null;
     boolean commited = false;
     try {
       openTransaction();
```

Comment:
Why do you need to wrap this in transaction? getMDatabase() already has transaction internally